core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -70,13 +74,34 @@ private ResourceReaper() {
         dockerClient = DockerClientFactory.instance().client();
     }
 
-    @SneakyThrows(InterruptedException.class)
+    /**
+     *
+     * @deprecated not for public usage
+     */
+    @Deprecated
     public static String start(String hostIpAddress, DockerClient client) {
+        return start(hostIpAddress, client, null);
+    }
+
+    /**
+     *
+     * @deprecated not for public usage
+     */
+    @Deprecated
+    @SneakyThrows(InterruptedException.class)
+    public static String start(String hostIpAddress, DockerClient client, @Nullable TransportConfig transportConfig) {
         String ryukImage = TestcontainersConfiguration.getInstance().getRyukImage();
         DockerClientFactory.instance().checkAndPullImage(client, ryukImage);
 
+        String socketPath = "//var/run/docker.sock";
+        if (transportConfig != null) {
+            URI dockerHost = transportConfig.getDockerHost();
+            if ("unix".equals(dockerHost.getScheme())) {
+                socketPath = dockerHost.getRawPath();
+            }
+        }
```

Comment:
I'd be inclined to add a new method to the `TransportConfig` object to perform this logic.

I'll start that change in #1771, as it's going to be necessary there, but FYI.

Reply:
@rnorth I am not sure whether this logic belongs to `TransportConfig`. Could you please elaborate why you want it there? :)

Reply:
In practice I'm not finding it fits there either 😄 
The strategy class works better